code/modules/wiremod/shell/module.dm

```diff
@@ -90,22 +93,41 @@
 	/// A reference to the wearer of the MODsuit
 	var/datum/port/output/wearer
 
+	/// Whether or not the suit is deployed
+	var/datum/port/output/deployed
+
+	/// Whether or not the suit is activated
+	var/datum/port/output/activated
+
 	/// The name of the last selected module
 	var/datum/port/output/selected_module
 
 	/// The signal that is triggered when a module is selected
 	var/datum/port/output/on_module_selected
 
+	/// The signal that is triggered when the suit is deployed by a signal
+	var/datum/port/output/on_deploy
+
+	/// The signal that is triggered when the suit has finished toggling itself after being activated by a signal
+	var/datum/port/output/on_toggle_finish
+
+/obj/item/circuit_component/mod_adapter_core/populate_options()
+	module_to_select = add_option_port("Module to Select", list())
+
 /obj/item/circuit_component/mod_adapter_core/populate_ports()
 	// Input Signals
-	module_to_select = add_input_port("Module to Select", PORT_TYPE_STRING)
+	toggle_deploy = add_input_port("Toggle Deployment", PORT_TYPE_SIGNAL)
 	toggle_suit = add_input_port("Toggle Suit", PORT_TYPE_SIGNAL)
 	select_module = add_input_port("Select Module", PORT_TYPE_SIGNAL)
 	// States
 	wearer = add_output_port("Wearer", PORT_TYPE_ATOM)
+	deployed = add_output_port("Deployed", PORT_TYPE_NUMBER)
+	activated = add_output_port("Activated", PORT_TYPE_NUMBER)
 	selected_module = add_output_port("Selected Module", PORT_TYPE_STRING)
 	// Output Signals
 	on_module_selected = add_output_port("On Module Selected", PORT_TYPE_SIGNAL)
+	on_deploy = add_output_port("On Deploy", PORT_TYPE_SIGNAL)
+	on_toggle_finish = add_output_port("Finished Toggling", PORT_TYPE_SIGNAL)
 
 /obj/item/circuit_component/mod_adapter_core/register_shell(atom/movable/shell)
```

Comment:
in this proc, `RegisterSignal(attached_module, COMSIG_MOVABLE_MOVED, .proc/on_move)` needs to be in the if statement block and not outside of it.

Reply:
I'm probably not even the one who fucked that one up, but yeah.

Reply:
yeah, but it's a bug nonetheless that I noticed